plugins/test/dynamic_test.cc

```diff
@@ -278,17 +279,42 @@ void DynamicTest::TestBody() {
     return;                                   \
   }
 
+void DynamicTest::EmitStats(benchmark::State& state,
+                            proxy_wasm::PluginHandleBase& handle,
+                            const TestContext& context) {
+  state.counters["WasmMemoryB"] = benchmark::Counter(
+      handle.wasm()->wasm_vm()->getMemorySize(), benchmark::Counter::kDefaults,
+      benchmark::Counter::kIs1024);
+  state.counters["LogsSizeB"] =
+      benchmark::Counter(context.logging_bytes(), benchmark::Counter::kDefaults,
+                         benchmark::Counter::kIs1024);
+  state.counters["WasmSizeB"] = benchmark::Counter(
+      std::filesystem::file_size(env_.wasm_path()),
+      benchmark::Counter::kDefaults, benchmark::Counter::kIs1024);
+}
+
 void DynamicTest::BenchPluginLifecycle(benchmark::State& state) {
   // Initialize VM.
   auto load_wasm = LoadWasm(/*benchmark=*/true);
   BM_RETURN_IF_ERROR(load_wasm.status());
   auto handle = *load_wasm;
 
   // Benchmark plugin initialization and teardown.
+  bool first = true;
   for (auto _ : state) {
     // Create root context (start) and call configure on it.
     auto plugin_init = InitializePlugin(handle);
     BM_RETURN_IF_ERROR(plugin_init);
+
+    // Emit root stats before teardown. After teardown the context is gone.
```

Comment:
This means we won't accurately reflect any logs in the OnDone/OnDelete callbacks for the plugin or VM context. That doesn't seem ideal, but it could be passable behavior. Thoughts?

Alternative seems to be owning logging_bytes_ outside of the `TestContext`, possibly in `TestWasm`?